include/sdf/Error.hh

```diff
@@ -179,6 +179,9 @@ namespace sdf
 
     /// \brief Generic error during parsing.
     PARSING_ERROR,
+
+    /// \brief File not found error.
+    FILE_NOT_FOUND,
```

Comment:
a review thread in https://github.com/gazebosim/sdformat/pull/1296#discussion_r1303216024 recommends using `FILE_READ` instead of adding a new enum for this

Reply:
I've opened https://github.com/gazebosim/sdformat/pull/1442 targeting this branch to use `FILE_READ` instead of a new enum